### PR TITLE
Updated go to 1.16. Added go modules support and golangci-lint.

### PR DESCRIPTION
**Pull request by jakefhyde**
~Leaving as draft for now until clarification on more specific improvements.~

Environment
--
version: `v2.6-head`
setup: `HA`

Changes
--
- Removed vendor support and added go modules.
- Updated to go 1.16.
- Replaced gometalinter with golangci-lint.
- Bump all dependencies.

Output
--
```
time="2021-11-08T22:48:01Z" level=info msg="Processing v1Authenticate request..."
time="2021-11-08T22:48:01Z" level=info msg=" ...looking up token for kubeconfig-user-52dng5wphl"
time="2021-11-08T22:48:01Z" level=info msg="{\"apiVersion\":\"authentication.k8s.io/v1beta1\",\"kind\":\"TokenReview\",\"status\":{\"authenticated\":true,\"user\":{\"username\":\"user-52dng\"}}}"
time="2021-11-08T22:48:01Z" level=info msg=" ...authenticated kubeconfig-user-52dng5wphl!"
```

Related Issue: https://github.com/rancher/rancher/issues/33332

### CODE REVIEW

File: Dockerfile.dapper [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
+
+RUN if [ "${ARCH}" == "amd64" ]; then \
+        curl -sL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.41.1; \
+    fi
 
 ENV DAPPER_ENV REPO TAG DRONE_TAG
 ENV DAPPER_SOURCE /go/src/github.com/rancher/kube-api-auth/
 ENV DAPPER_OUTPUT ./bin ./dist
 ENV DAPPER_DOCKER_SOCKET true
+ENV DAPPER_RUN_ARGS "-v rancher2-go16-pkg-1:/go/pkg -v rancher2-go16-cache-1:/root/.cache/go-build --privileged"
```
Comment (thedadams) [inline Dockerfile.dapper]: Is the `--privileged` flag necessary?

File: pkg/service/controllers/setup.go [hunk 2]
```diff
@@ -3,14 +3,9 @@ package controllers
...
 )
 
-func Start(ctx context.Context, namespace string, apiContext *config.UserOnlyContext) error {
-	return controller.SyncThenStart(ctx, 1,
-		apiContext.Cluster.ClusterAuthTokens(namespace).Controller(),
-		apiContext.Cluster.ClusterUserAttributes(namespace).Controller(),
-		apiContext.Core.ConfigMaps(namespace).Controller(),
-	)
+func Start(ctx context.Context, apiContext *config.UserOnlyContext) error {
+	return apiContext.Start(ctx)
```
Comment (thedadams) [inline pkg/service/controllers/setup.go:10]: This seems like a significant change unrelated to adding go module support. Can you explain what's going on here? Why are we getting rid of the `SyncThenStart`?

**Review by thedadams — COMMENTED** [review]
Just a couple questions here.

File: pkg/service/controllers/setup.go (hunk 2 above)
Comment (jakefhyde) [reply pkg/service/controllers/setup.go:10]: `SyncThenStart` was removed from norman sometime after 2.4, and 2.4 couldn't be used without some difficulty (`Starter` interface was completely removed from the controllers that we are using in rancher/rancher and was required for `SyncThenStart`). The generated go.mod from the vendor.conf was invalid, so dependencies would have had to be selectively updated to find compatible versions which seemed like too much effort to not update.

File: Dockerfile.dapper (hunk 1 above)
Comment (jakefhyde) [reply Dockerfile.dapper]: Good catch, no I don't think so, think I left that in when copying from rancher. Updating.

**Review by paynejacob — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Pull request merged**